### PR TITLE
Add basic, chroma, pinecone retrievers and related tests

### PR DESCRIPTION
**Pull request by umutunsal**

### CODE REVIEW

File: hive_agent/tools/retriever/base_retrieve.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    def create_basic_index(self, file_path=None, folder_path=None):
+        documents = self._load_documents(file_path, folder_path)
+        index = VectorStoreIndex.from_documents(documents)
+        return index
+
+    def insert_documents(self, index, file_path=None, folder_path=None):
+        documents = self._load_documents(file_path, folder_path)
+        if not documents:
+            return "No documents found to insert."
```
Comment (ToJen) [inline hive_agent/tools/retriever/base_retrieve.py:42]: should we raise an error with the error message here instead?

File: hive_agent/tools/retriever/pinecone_retrieve.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+class PineconeRetriever(RetrieverBase):
+
+    def __init__(self):
+        PINECONE_API_KEY = os.environ["PINECONE_API_KEY"]
+        super().__init__(
+            name="PineconeRetriever",
+            description="This tool creates a Pinecone Retriever index",
+        )
+        self.pinecone_client = Pinecone(api_key=PINECONE_API_KEY)
```
Comment (ToJen) [inline hive_agent/tools/retriever/pinecone_retrieve.py:20]: add API key to **.env.example** file

**Review by ToJen — CHANGES_REQUESTED** [review]

**Comment by umutunsal** [comment]
More discussed topics
- Create basic index automatically when /upload documents endpoint called
- Get a parameter to control index store types like full text, hybrid, semantic etc.

**Comment by umutunsal** [comment]
Except "Get a parameter to control index store types like full text, hybrid, semantic etc." this feature all of them fixed or implemented. Llamaindex chroma integration "stores_text" parameter is not working properly and needs more deep dive.

**Review by ToJen — DISMISSED** [review]

**Review by ToJen — APPROVED** [review]

**Review by ToJen — APPROVED** [review]

**Pull request merged**